virttest/utils_disk.py

```diff
@@ -182,6 +190,559 @@ def clean_old_image(image):
+    :param fstype: file system type for the disk.
+    """
+    disk = "disk_" + ''.join(random.sample(string.ascii_letters + string.digits, 4))
+    format_cmd = "echo select volume %s > " + disk
+    format_cmd += " && echo format fs=%s quick >> " + disk
+    format_cmd += " && echo exit >> " + disk
+    format_cmd += " && diskpart /s " + disk
+    format_cmd += " && del /f " + disk
+    session.cmd(format_cmd % (partition_name, fstype), timeout=360)
+
+
+def create_filesystem(session, partition_name, fstype, ostype):
+    """
+    create file system in windows or linux guest.
+
+    :param session: session object to guest.
+    :param partition_name: partition name that to be formatted.
+    :param fstype: file system type for the disk.
+    :param ostype: guest os type 'windows' or 'linux'.
+    """
+    if ostype == "windows":
+        create_filesystem_windows(session, partition_name, fstype)
+    else:
+        create_filesyetem_linux(session, partition_name, fstype)
+
+
+def set_drive_letter(session, did, partition_no=1):
+    """
+    set drive letter for partition in windows guest.
+
+    :param session: session object to guest.
+    :param did: disk index
+    :param partition_no: partition number
+    :return drive_letter: drive letter has been set
+    """
+    drive_letter = ""
+    list_partition_cmd = ' echo list partition '
+    list_partition_cmd = _wrap_windows_cmd(list_partition_cmd)
+    details = session.cmd_output(list_partition_cmd % did)
+    for line in details.splitlines():
+        if re.search("Reserved", line, re.I | re.M):
+            partition_no += int(line.split()[1])
+    assign_letter_cmd = ' echo select partition %s ; echo assign '
+    assign_letter_cmd = _wrap_windows_cmd(assign_letter_cmd)
+    session.cmd(assign_letter_cmd % (did, partition_no))
+    detail_cmd = ' echo detail disk '
+    detail_cmd = _wrap_windows_cmd(detail_cmd)
+    details = session.cmd_output(detail_cmd % did)
+    for line in details.splitlines():
+        pattern = "\s+Volume\s+\d+"
+        if re.search(pattern, line, re.I | re.M):
+            drive_letter = line.split()[2]
+            break
+    if len(drive_letter) == 1 and drive_letter.isalpha():
+        return drive_letter
+    else:
+        return None
+
+
+def drop_drive_letter(session, drive_letter):
+    """
+    remove drive letter for partition in windows guest.
+
+    :param session: session object to guest.
+    :param drive_letter: drive letter to be removed
+    """
+    disk = "disk_" + ''.join(random.sample(string.ascii_letters + string.digits, 4))
+    remove_cmd = "echo select volume %s > " + disk
+    remove_cmd += " && echo remove >> " + disk
+    remove_cmd += " && echo exit >> " + disk
+    remove_cmd += " && diskpart /s " + disk
+    remove_cmd += " && del /f " + disk
+    session.cmd(remove_cmd % drive_letter)
+
+
+def format_windows_disk(session, did, size, n_partitions=1, fstype="ntfs",
```

Comment:
I'd suggest using the function name `configure_empty_$ostype_disk()`.

Reply:
updated